File: C#/rna-transcription/RnaTranscription.cs
//  Date: 10-25-19
//  Exercism IO RNA Transciption
//  Purpose: receive a DNA input string and transcribe it to the corresponding RNA code

using System.Text;

public static class RnaTranscription
{
    public static string ToRna(string nucleotide)
    {
        string DNA = "ACGT";
        string RNA = "UGCA";
        StringBuilder sb = new StringBuilder(nucleotide.Length);

        foreach (char s in nucleotide)
        {
            int index = DNA.IndexOf(s);
            sb.Append(RNA[index]);
        }
        
        return sb.ToString();
    }
}

File: JavaScript/rna-transcription.js
//
// This is only a SKELETON file for the 'RNA Transcription' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const toRna = (dna) => {
  let DNA = 'ACGT';
  let RNA = 'UGCA';
  let result = '';
  // take each letter in input string and use [...] to turn into array. find the index in DNA and add the same index
  // RNA to result string
  [...dna].forEach(letter => result += RNA[DNA.indexOf(letter)]);
  return result;
};
